packages/shield-controller/src/backend.test.ts

```diff
@@ -250,5 +335,295 @@ describe('ShieldRemoteBackend', () => {
         }),
       ).rejects.toThrow('Failed to log transaction: 500');
     });
+
+    it('aborts pending coverage result polling before logging', async () => {
+      const { backend, fetchMock } = setup({
+        getCoverageResultTimeout: 10000,
+        getCoverageResultPollInterval: 100,
+      });
+
+      const txMeta = generateMockTxMeta();
+      const coverageId = 'coverageId';
+
+      // Setup fetch to handle init, result polling, and log requests
+      fetchMock.mockImplementation(
+        async (input: RequestInfo | URL, init?: RequestInit) => {
+          // eslint-disable-next-line @typescript-eslint/no-base-to-string
+          const url = typeof input === 'string' ? input : input.toString();
+          const signal = init?.signal;
+
+          if (url.includes('/init')) {
+            return {
+              status: 200,
+              json: jest.fn().mockResolvedValue({ coverageId }),
+            } as unknown as Response;
+          }
+
+          if (url.includes('/result')) {
+            if (signal?.aborted) {
+              throw new Error('Request was aborted');
+            }
+            // Keep polling (never return 200) to test abortion
+            return {
+              status: 412,
+              json: jest.fn().mockResolvedValue({ status: 'unknown' }),
+            } as unknown as Response;
+          }
+
+          if (url.includes('/log')) {
+            return { status: 200 } as unknown as Response;
+          }
+
+          throw new Error('Unexpected URL');
+        },
+      );
+
+      // Start coverage check (don't await) - this will start polling
+      const coveragePromise = backend.checkCoverage({ txMeta });
+
+      // Wait a bit to let polling start
+      await delay(50);
+
+      // Log transaction - this should abort the ongoing polling
+      await backend.logTransaction({
+        txMeta,
+        transactionHash: '0x00',
+        status: 'shown',
+      });
+
+      // Coverage check should be cancelled
+      await expect(coveragePromise).rejects.toThrow(
+        'Coverage result polling cancelled',
+      );
+    });
+  });
+
+  // Testing coverage result polling with timeout and cancellation.
+  describe('withTimeoutAndCancellation', () => {
+    it('should abort previous result request when new request is made', async () => {
+      const { backend, fetchMock } = setup({
+        getCoverageResultTimeout: 10000, // Long timeout to avoid timeout during test
+        getCoverageResultPollInterval: 100,
+      });
+
+      const signatureRequest = generateMockSignatureRequest();
+      const coverageId = 'coverageId';
+
+      // Mock `/init` and `/result` responses.
+      fetchMock.mockImplementation(
```

Comment:
This mock implementation is in parts redundant to the other fetchMock implementations.